daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/SequenceChild.scala

```diff
@@ -27,60 +27,294 @@ import org.apache.daffodil.schema.annotation.props.gen.OccursCountKind
 
-  final override def parser = sequenceChildParser
-  final override def unparser = sequenceChildUnparser
+  final override lazy val parser = sequenceChildParser
+  final override lazy val unparser = sequenceChildUnparser
 
   protected def sequenceChildParser: SequenceChildParser
   protected def sequenceChildUnparser: SequenceChildUnparser
 
-  def optSequenceChildParser: Option[SequenceChildParser] =
+  final lazy val optSequenceChildParser: Option[SequenceChildParser] =
     if (childParser.isEmpty) None else Some(parser)
 
-  def optSequenceChildUnparser: Option[SequenceChildUnparser] =
+  final lazy val optSequenceChildUnparser: Option[SequenceChildUnparser] =
     if (childUnparser.isEmpty) None else Some(unparser)
 
-  protected lazy val sepGram = sq.sequenceSeparator
+  /**
+   * There's only parse result helpers here, so let's abbreviate
+   */
+  protected type SeparatedHelper = SeparatedSequenceChildParseResultHelper
+  protected type UnseparatedHelper = UnseparatedSequenceChildParseResultHelper
+
+  protected def separatedHelper: SeparatedHelper
+  protected def unseparatedHelper: UnseparatedHelper
+
+  protected lazy val sepGram = {
+    sscb match {
+      case _: PositionalLike =>
+        sgtb.checkSeparatorTerminatorConflict
+      case _ => // ok
+    }
+    sq.sequenceSeparator
+  }
+
   protected lazy val sepParser = sepGram.parser
   protected lazy val sepUnparser = sepGram.unparser
 
-  lazy val srd = sq.sequenceRuntimeData
-  lazy val trd = child.termRuntimeData
+  final protected def srd = sq.sequenceRuntimeData
+  final protected def trd = child.termRuntimeData
+  final def termRuntimeData = trd
+  final protected lazy val mgrd = child.asInstanceOf[ModelGroup].modelGroupRuntimeData
+  final protected lazy val erd = child.asInstanceOf[ElementBase].elementRuntimeData
+
+  private lazy val childElement = child.asInstanceOf[ElementBase]
+  private lazy val childModelGroup = child.asInstanceOf[ModelGroup]
+
+  protected final def ssp = sq.separatorSuppressionPolicy
+
+  protected lazy val separatedSequenceChildBehavior: SeparatedSequenceChildBehavior = {
+    import SeparatorSuppressionPolicy._
+    import SeparatedSequenceChildBehavior._
+    child match {
+      case m: ModelGroup => {
+        if (child.isPotentiallyTrailing) {
+          ssp match {
+            case AnyEmpty => NonPositional
+            case TrailingEmpty => PositionalTrailingLax
+            case TrailingEmptyStrict => PositionalTrailingStrict
+            case Never => Positional
+          }
+        } else {
+          Positional
+        }
+      }
+      case eb: ElementBase if (eb.isArray || eb.isOptional) => {
+        import OccursCountKind._
+        eb.occursCountKind match {
+          case Fixed => Positional
+          case Expression => Positional
+          case Parsed => NonPositional
+          case StopValue => NonPositional
+          case Implicit => {
+            val UNB = -1
+            (
```

Comment:
I know this was probably a result of your formatting tool settings, but this lone parens looks really weird to me.

Reply:
Yeah. Looks odd to me too, but scalariform firstArgumentOnNewline defaults to Force, which says that if the args to a function call don't all fit on one line, then the first goes on a new line as do all the others.

In this case the function being called is the tuple constructor, so it seems odd, because we don't even think of these tuples as objects being constructed. But I think this is the firstArgumentOnNewline behavior.